libdnf/rpm/solv/package_private.hpp

```diff
@@ -361,6 +361,16 @@ inline Repo * get_repo(Pool * pool, libdnf::rpm::PackageId package_id) noexcept
     return static_cast<Repo *>(solvable->repo->appdata);
 }
 
+inline Id id_to_lowercase_id(Pool * pool, Id id_input) {
+    auto name_cstring = pool_id2str(pool, id_input);
+    int name_length = static_cast<int>(strlen(name_cstring));
+    auto tmp_name_cstring = pool_alloctmpspace(pool, name_length);
+    for (int index = 0; index < name_length; ++index) {
+        tmp_name_cstring[index] = static_cast<char>(tolower(static_cast<unsigned char>(name_cstring[index])));
```

Comment:
Why are you cast `char` to `unsigned char` (`tolower(static_cast<unsigned char>(name_cstring[index]))`)?
`tolower` takes `int`. So, `tolower(static_cast<int>(name_cstring[index]))` is better.
Or simply `tolower(name_cstring[index])` without casting.